app/models/customer.rb

```diff
@@ -0,0 +1,5 @@
+class Customer < ApplicationRecord
+  validates_presence_of :name
+
+  has_and_belongs_to_many :items, join_table: 'customer_items'
```

Comment:
What we've taught you so far with many-to-many-tables doesn't always work perfectly with this has_and_belongs_to_many association. That association is the same idea but if you use that association, your joins table can't have any additional fields besides the foreign key. We want the flexibility to be able to add columns to our joins tables as needed (for example, adding an additional status column to the PetApplications table in the last project). So, we teach students the has_many :items, through :customer_items instead of this association so that you have more flexibility.

tl;dr: while this association can be handy in some cases, use the association `has_many .. through:` instead because it's more flexible in all cases.

Reply:
For this case though, this association works! But don't want you to try to use it in other situations where it won't work which is why we prefer `has_many .. through:`